src/OneBeyond.Studio.Infrastructure.Azure/MessageQueues/StorageMessageQueueBase.cs

```diff
@@ -22,7 +22,7 @@ protected StorageMessageQueueBase(AzureMessageQueueOptions options)
         options.EnsureIsValid();
 
         _messageQueue = new AsyncLazy<QueueClient>(
-            () => Queue.GetOrCreateAsync(options.ConnectionString!, options.QueueName!), //We ensured both ConnectionString and QueueName are specified above
+            () => Queue.GetOrCreateAsync(options.ResourceName, options.ConnectionString!, options.QueueName!),
```

Comment:
This means that the `ConnectionString` is mandatory even when we provide the `ResourceName`? How about if we use something like `UseAzureCredentials` to decide which one to look for?

Reply:
Fixed